### PR TITLE
S2 editable student profile t5 create update student profile js

### PR DESCRIPTION
**Pull request by chloe-smart**
- Created the JS file, made functionality that works for everything, but only applied it to the "edaid" section 
- Modified code in DisplayStudentViewHelper to add demo edit button to check the JS code works (which it does!)
- Need to uncomment the other event listeners in JS file when we have t4 code & replace the demo code in  DisplayStudentViewHelper with t4 code
- For testing also added a script tag in studentProfile.phtml file (but that won't cause any merge conflicts)

### CODE REVIEW

File: public/js/updateStudentProfile.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+const laptopEditButton = document.querySelector('.laptopEditButton')
+const githubUserEditButton = document.querySelector('.githubUserEditButton')
+
+function handleEditClick(event) {
+    const buttonName = event.target.getAttribute('class')
+    const divName = buttonName.replace('EditButton', 'Container')
+    const divSelector = document.querySelector('.' + divName)
+    divSelector.innerHTML = 
+        '<form method="put">' +
+        '<label for="textBox">EdAid amount: </label>' +
```
[the remark this answers is not in the document]
Comment (tauroid) [reply public/js/updateStudentProfile.js:12]: or actually divSelector.childNodes[0].textContent will get you the heading

File: public/js/updateStudentProfile.js [hunk 2]
```diff
@@ -0,0 +1,23 @@
+const edaidEditButton = document.querySelector('.edaidEditButton')
+const upfrontEditButton = document.querySelector('.upfrontEditButton')
+const laptopEditButton = document.querySelector('.laptopEditButton')
+const githubUserEditButton = document.querySelector('.githubUserEditButton')
+
+function handleEditClick(event) {
+    const buttonName = event.target.getAttribute('class')
```
Comment (tauroid) [inline public/js/updateStudentProfile.js:7]: slightly weird but sure

File: public/js/updateStudentProfile.js [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+    const divName = buttonName.replace('EditButton', 'Container')
+    const divSelector = document.querySelector('.' + divName)
```
Comment (tauroid) [inline public/js/updateStudentProfile.js:9]: it's not a div selector, it's just the div. so e.g. containerDiv

**Review by tauroid — CHANGES_REQUESTED** [review]

**Review by tauroid — CHANGES_REQUESTED** [review]
comments

File: public/js/updateStudentProfile.js (hunk 3 above)
Comment (chloe-smart) [reply public/js/updateStudentProfile.js:9]: done👍

File: public/js/updateStudentProfile.js [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+    divSelector.innerHTML = 
+        '<form method="put">' +
+        '<label for="textBox">EdAid amount: </label>' +
+        '<input type="text" id="textBox" name="' + divName + '">' +
```
[the remark this answers is not in the document]
Comment (chloe-smart) [reply public/js/updateStudentProfile.js:13]: done 👍

File: public/js/updateStudentProfile.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (chloe-smart) [reply public/js/updateStudentProfile.js:12]: done thanks!🥳

File: src/ViewHelpers/DisplayStudentProfileViewHelper.php [hunk 5]
```diff
@@ -51,7 +51,13 @@ public static function outputApplicant(CompleteApplicantEntity $applicant): stri
                     <h4>Onboarding Info</h4>
                     <p class="detail">Diversitech amount: <span id="diversitech">'
                         . $applicant->getDiversitech() . '</span>
-                    <p class="detail">EdAid amount: <span id="edaid">' . $applicant->getEdaid() . '</span></p>
+
+                    <!-- DEMO EDIT BUTTON, REPLACE WITH TASK 4 CODE --> 
+                    <div class="detail edaidContainer">EdAid amount: 
+                        <span id="edaid">' . $applicant->getEdaid() . '</span>
+                        <button class="edaidEditButton">Edit</button>
+                    </div>
```
[the remark this answers is not in the document]
Comment (chloe-smart) [reply src/ViewHelpers/DisplayStudentProfileViewHelper.php:59]: 🙃

File: public/js/updateStudentProfile.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (chloe-smart) [reply public/js/updateStudentProfile.js:13]: done👍

**Pull request merged**